src/traces/contour/attributes.js

```diff
@@ -44,13 +44,15 @@ module.exports = extendFlat({}, {
     },
     ncontours: {
         valType: 'integer',
-        dflt: 0,
+        dflt: 15,
+        min: 1,
         role: 'style',
         description: [
             'Sets the maximum number of contour levels. The actual number',
             'of contours will be chosen automatically to be less than or',
             'equal to the value of `ncontours`.',
-            'Has an effect only if `autocontour` is *true*.'
+            'Has an effect only if `autocontour` is *true* or if',
+            '`contours.size` is missing.'
```

Comment:
Hmm, I was starting to think `autocontour` had become completely obsolete, we’d just fill in whatever missing pieces there were. But actually I think I want to make some changes to this:

- with `autocontour=true` we clear `start`, `end`, and `size` and generate them all automatically from the data range and `ncontours`. Perhaps the only reason you'd need to do this is if you previously set some non-auto values and you want to toss them out.
- with `autocontour=false` we still fill in anything that’s missing but won’t ignore anything the user provides. So for example if you only want to specify `contours.end` to ignore some outliers, you can do that, you don’t have to also specify `contours.start` to get it accepted.

I don't think there's anything backward-incompatible in this proposal, except in partial information cases, where we had a bunch of bugs anyway.